lib/tutorials/caching.md

```diff
@@ -0,0 +1,187 @@
+
+You only need to set `ETag` in your handler via `etag(tag, options)` function:
+
+```
+   reply(result).etag('xxxxxxxxx');
+```
+
+Check `options` parameter for `etag` function on [api](http://hapijs.com/api#response-object) page for more details.
+
+
+## Server side caching
+
+hapi provides powerful server side caching via [catbox](https://www.github.com/hapijs/catbox) and makes using cache very convenient. This tutorial help you to understand how catbox is utilized inside hapi and how you can use it. Catbox also can be used out of the hapi, but that is not covered here.
+
+Catbox has two interfaces.
+
+
+### Client
+[Client](https://github.com/hapijs/catbox#client) is low-level interface, which basically allows you set/get key-value pairs. Its initialized with one of the available adapters ([Memory](https://github.com/hapijs/catbox-memory), [Redis](https://github.com/hapijs/catbox-redis), [mongoDB](https://github.com/hapijs/catbox-mongodb), [Memcached](https://github.com/hapijs/catbox-memcached), [Riak](https://github.com/DanielBarnes/catbox-riak), [Amazon S3](https://github.com/fhemberger/catbox-s3)).
+
+hapi always initialize one default [client](https://github.com/hapijs/catbox#client) with [memory](https://github.com/hapijs/catbox-memory) adapter. Lets see how we can define more clients.
+
+```javascript
+var Hapi = require('hapi');
+
+var server = new Hapi.Server({
+    cache: [
+        {
+            name: 'mongoCache',
+            engine: require('catbox-mongodb'),
+            host: '127.0.0.1',
+            partition: 'cache',
+        },
+        {
+            name: 'redisCache',
+            engine: require('catbox-redis'),
+            host: '127.0.0.1',
+            partition: 'cache'
+       }
+    ]
+});
+
+server.connection({
+    port: 8000
+});
+```
+
+In this example two [clients](https://github.com/hapijs/catbox#client) were defined: `'mongoCache'`, `'redisCache'` and also default one without name with [memory](https://github.com/hapijs/catbox-memory) adapter. You can replace default client by omitting `name` property. `partition` property tells adapter how cache should be named (`'catbox'` by default). In case of [mongoDB](http://www.mongodb.org/) it is database name and in case of [redis](http://redis.io/) it is used as key prefix.
+
+### Policy
+[Policy](https://github.com/hapijs/catbox#policy) is high-level interface that you will be using. Lets pretend we are dealing with something more complicated than adding two numbers and we want to cache the results. [server.cache](http://hapijs.com/api#servercacheoptions) method creates new [policy](https://github.com/hapijs/catbox#policy), which is than used in handler. `id` object has mandatory key `id`, which is a unique item identifier string.
```

Comment:
Maybe remove the last sentence and create a new paragraph explaining `sumCache.get` below the code. 

It might be worth explaining what the `generateFunc` does (like why the signature is the way it is) as well.